modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/nhttp/NHttpConfiguration.java

```diff
@@ -203,6 +199,41 @@ public int getListenerShutdownWaitTime() {
     }
 
     /**
+     * Check preserving status of the http header field
+     *
+     * @param httpHeader http header name
+     * @return return true if preserve else false
+     */
+    public boolean isPreserveHttpHeader(String httpHeader) {
+        if (preserveHeaders == null) {
+            preserveHeaders = new ArrayList<String>();
```

Comment:
We need to move this initialization logic into the constructor. Otherwise first request will take some more time to read files and load properties.  We need to make sure that we are not putting any logic in the request flow that can be done prior to request processing.